File: app/src/main/java/com/blueskybone/arkscreen/base/preference/shared/SharedPreferenceStore.java
package com.blueskybone.arkscreen.base.preference.shared;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import com.blueskybone.arkscreen.base.preference.Preference;
import com.blueskybone.arkscreen.base.preference.PreferenceStore;

import java.util.Set;
import java.util.function.Function;

/*
 * The code for data persistence refers to the following project
 * https://github.com/easybangumiorg/EasyBangumi
 * */

public class SharedPreferenceStore implements PreferenceStore {

    public SharedPreferenceStore(Context context) {
        this.sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
    }

    private final SharedPreferences sharedPreferences;

    @Override
    public Preference<String> getString(String key, String value) {
        return new SharedPreference.StringPrimitive(sharedPreferences, key, value);
    }

    @Override
    public Preference<Integer> getInt(String key, int value) {
        return new SharedPreference.IntegerPrimitive(sharedPreferences, key, value);
    }

    @Override
    public Preference<Long> getLong(String key, Long value) {
        return new SharedPreference.LongPrimitive(sharedPreferences, key, value);
    }

    @Override
    public Preference<Float> getFloat(String key, Float value) {
        return new SharedPreference.FloatPrimitive(sharedPreferences, key, value);
    }

    @Override
    public Preference<Boolean> getBoolean(String key, Boolean value) {
        return new SharedPreference.BooleanPrimitive(sharedPreferences, key, value);
    }

    @Override
    public <T> Preference<T> getObject(String key, T defaultValue, Function<T, String> serializer, Function<String, T> deserializer) {
        return new SharedPreference.Object<>(sharedPreferences, key, defaultValue, serializer, deserializer);
    }

    @Override
    public Preference<Set<String>> getStringSet(String key, Set<String> set) {
        return new SharedPreference.StringSetPrimitive(sharedPreferences, key, set);
    }
}

File: app/src/main/java/com/blueskybone/arkscreen/preference/preference/shared/SharedPreferenceStore.kt
package com.blueskybone.arkscreen.preference.preference.shared

import android.content.Context
import android.content.SharedPreferences
import com.blueskybone.arkscreen.preference.preference.Preference
import com.blueskybone.arkscreen.preference.preference.PreferenceStore
import java.util.function.Function

/**
 *   Created by blueskybone
 *   Date: 2025/1/7
 */
class SharedPreferenceStore(context: Context) : PreferenceStore {

    companion object{
        private lateinit var sharedPreferences: SharedPreferences
    }
    init{
        sharedPreferences = context.getSharedPreferences("context",Context.MODE_PRIVATE)
    }

    override fun getString(key: String, value: String): Preference<String> {
        return SharedPreference.StringPrimitive(sharedPreferences, key, value)
    }

    override fun getInt(key: String, value: Int): Preference<Int> {
        return SharedPreference.IntegerPrimitive(sharedPreferences, key, value)
    }

    override fun getLong(key: String, value: Long): Preference<Long> {
        return SharedPreference.LongPrimitive(sharedPreferences, key, value)
    }

    override fun getFloat(key: String, value: Float): Preference<Float> {
        return SharedPreference.FloatPrimitive(sharedPreferences, key, value)
    }

    override fun getBoolean(key: String, value: Boolean): Preference<Boolean> {
        return SharedPreference.BooleanPrimitive(sharedPreferences, key, value)
    }

    override fun <T> getObject(
        key: String,
        defaultValue: T,
        serializer: Function<T, String>,
        deserializer: Function<String, T>
    ): Preference<T> {
        return SharedPreference.Object(
            sharedPreferences,
            key,
            defaultValue,
            serializer,
            deserializer
        )
    }


}